RESTResource/RESTResource.js

```diff
@@ -289,6 +289,27 @@ export default class RESTResource {
     return actions;
   }
 
+  // Some options properties may be an object or a function, determine which
+  // the passed property is and act accordingly
+  buildOption = (option, props, state) => {
+    let toReturn;
+    if (typeof option === 'object') {
+      toReturn = _.mapValues(
+        option,
+        param => substitute(param, props, state, this.module, this.logger, this.dataKey)
+      );
+      for (const key of Object.keys(option)) {
+        if (option[key] === null) {
+          return null;
+        }
+      }
```

Comment:
```suggestion
      const someParamIsNull = Object.values(option).some(value => value === null)
      if (someParamIsNull) return null; // or toReturn = null

```
Is that the point here, return `null` if some of the params is null?

Reply:
> Is that the point here, return null if some of the params is null?

That seems to be the intent of that portion of code, however, it isn't a result of my refactor, it was just part of the pre-existing code that was refactored.

I'm not sure allowing the scope of my PR to creep by making arbitrary changes is a good idea, I'd rather keep the focus on the intention of my PR. I implemented the suggestion to switch from _.get to optional chaining following the comments from @MikeTaylor and @doytch since a move to deprecate _.get seems inevitable, but your change, although perfectly valid, seems like it's moving outside of the scope of the PR.

Happy to be shouted down though! :)

Reply:
I didn't mean to shout down )
In fact, I'm more concerned about not the syntax but the API - returning `null` if one of the params is exactly `null` seems strange, out of context.

Reply:
From what I can see, both the implementation in the current commit and the implementation that @axelhunn is suggesting are functionally equivalent. The only difference I see is short-circuiting the loop via `.some` instead of always running over every iteration of the object. The performance is not relevant here, but I think Alex's version does make it a bit clearer.

I totally understand any reticence to change the guts of the algorithm, but in this case I'd say you're well within the scope of the PR.

And yeah, the intention is to return `null` if anything computes to `null`. I think `null` is consciously selected here over undefined to attempt hammer home the point that the resource's owner purposely halted this fetch because it cannot be completed properly.

Reply:
@axelhunn Sorry, I didn't mean that you were shouting me down! I meant that I felt it was out of scope to make the change you were suggesting, but I was happy to do it if it was felt that the change should be included :+1:

As far as your question as to the sanity of returning null in the event that any of the params or headers is null, that's a good question.  It certainly seems to be intentional since @MikeTaylor [refactored it](https://github.com/folio-org/stripes-connect/commit/d8f91b1afd0d66884a61abd6fee1d6aef9aca40a)  at one point to keep Jenkins happy, and the [original commit](https://github.com/folio-org/stripes-connect/commit/31485ee969d1ccec4a2c01ed7962e177fd0a43d4) gives a bit of context.